mediatimestamp/immutable/timerange.py

```diff
@@ -794,3 +794,72 @@ def normalise(self,
+                while True:
+                    if last_tr is None:
+                        last_tr = self.from_start_length(
+                            self.start, time_duration, self.Inclusivity(self.includes_start()))
+                    else:
+                        last_tr = self.from_start_length(last_tr.end, time_duration, self.INCLUDE_START)
+                    yield last_tr
+            # The input time range is not infinite
+            remaining_length: int = cast(TimeOffset, self.length).to_nanosec() - time_duration.to_nanosec()
+            while remaining_length > 0:
+                if last_tr is None:  # This is the first and not the last
+                    # The first must always have the same start inclusivity as the timerange, and an exclusive ) end.
+                    # The four possibilities for inclusivity are defined in this file as binary integers 0-3.
+                    # The first bit is the end inclusivity - for exclusive ends the first bit must be zero, therefore
+                    # we only need to use the last bit which indicates the start inclusivity.
+                    last_tr = self.from_start_length(self.start, time_duration, self.Inclusivity(self.includes_start()))
+                else:  # this is not the first and not the last; it must have inclusive start and exclusive end
+                    # We can also continue on from the end of the last timerange for our start timestamp.
+                    if last_tr.end is not None:
+                        last_tr = self.from_start_length(last_tr.end, time_duration, self.INCLUDE_START)
+                    else:
+                        raise Exception("Something has gone badly wrong, the last chunk had no end value set!")
+                remaining_length -= time_duration.to_nanosec()
+                yield last_tr
+
+            # remaining_length will be 0 if this is an exact chunk divisor, or negative if we have a short final chunk.
+            # Either way, it's the final chunk.
+            # The sum of the requested duration and time remaining will determine the length of the final chunk.
+            length_of_last_chunk: int = time_duration.to_nanosec() + remaining_length
+            if last_tr is None:  # There is only one chunk, and it's an exact fit OR shorter than the requested dur.
+                yield self.from_start_length(self.start, TimeOffset(ns=length_of_last_chunk), self.inclusivity)
+            else:
+                if last_tr.end is not None:
+                    yield self.from_start_length(
+                        last_tr.end, TimeOffset(ns=length_of_last_chunk),
+                        self.Inclusivity((self.includes_end() << 1) + 1))
+                else:
+                    raise Exception("Something has gone badly wrong, the last chunk had no end value set!")
+        else:
+            raise Exception("This TimeRange object has no 'start' value and so cannot be chunked!")
```

Comment:
Mind making this a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError)? The others might make sense to be too - makes them easier to catch